### PR TITLE
🏗 [pr-deploy-bot] Make Travis send skipped or errored results if dist skipped or errored

### PR DESCRIPTION
**Pull request by estherkim**
Handle if dist is skipped or errors for the pr-deploy bot. Created a util function `timedExecWithError()` that doesn't kill the node process but exposes errors, if any.

### CODE REVIEW

File: build-system/pr-check/utils.js [hunk 1]
```diff
@@ -210,6 +207,22 @@ function timedExec(cmd, fileName = 'utils.js') {
   return p;
 }
 
+/**
+ * Executes the provided command and times it. Errors, if any, are returned.
+ * @param {string} cmd
+ * @param {string} fileName
+ * @return {<Object>} Process info.
```
Comment (danielrozenberg) [inline build-system/pr-check/utils.js]: Process info or error

File: build-system/tasks/pr-deploy-bot-utils.js [hunk 2]
```diff
@@ -16,12 +16,20 @@
...
+const {cyan, green} = require('ansi-colors');
 const {gitCommitHash} = require('../git');
 const {replaceUrls: replaceUrlsAppUtil} = require('../app-utils');
 const {travisBuildNumber} = require('../travis');
 
+const baseUrl = 'https://amp-pr-deploy-bot.appspot.com/v0/pr-deploy/';
+const results = new Map()
+  .set('success', 0)
+  .set('errored', 1)
+  .set('skipped', 2);
```
Comment (danielrozenberg) [inline build-system/tasks/pr-deploy-bot-utils.js]: Q: why not pass the status strings directly to the app, and have the app `switch` on those instead on converting numbers back and forth?

File: build-system/pr-check/dist-bundle-size.js [hunk 3]
```diff
@@ -28,25 +28,31 @@ const {
...
 const FILENAME = 'dist-bundle-size.js';
 const FILELOGPREFIX = colors.bold(colors.yellow(`${FILENAME}:`));
 const timedExecOrDie = (cmd, unusedFileName) =>
   timedExecOrDieBase(cmd, FILENAME);
 
+function stopJob(startTime) {
+  stopTimer(FILENAME, FILENAME, startTime);
+  process.exitCode = 1;
+  return;
+}
```
Comment (rsimha) [inline build-system/pr-check/dist-bundle-size.js]: Nice! As a follow up, maybe move this function to `utils.js` and replace [all these instances](https://github.com/ampproject/amphtml/search?utf8=%E2%9C%93&q=%22stopTimer(FILENAME%2C+FILENAME%2C+startTime)%22) too?

File: build-system/pr-check/utils.js [hunk 4]
```diff
@@ -210,6 +207,22 @@ function timedExec(cmd, fileName = 'utils.js') {
...
+ */
+function timedExecWithError(cmd, fileName = 'utils.js') {
+  const startTime = startTimer(cmd, fileName);
+  const p = exec(cmd, {'stdio': ['inherit', 'inherit', 'pipe']});
+  stopTimer(cmd, fileName, startTime);
+  if (p.stderr.length > 0) {
+    p.error = new Error(p.stderr.toString());
+  }
+  return p;
+}
```
Comment (rsimha) [inline build-system/pr-check/utils.js]: Would it make sense to add a function called `execWithError` to `build-system/exec.js`, and use it here?

File: build-system/pr-check/utils.js (hunk 1 above)
Comment (estherkim) [reply build-system/pr-check/utils.js]: I think Process here is a noun lol

File: build-system/exec.js [hunk 5]
```diff
@@ -71,6 +71,20 @@ function execOrDie(cmd, options) {
   }
 }
 
+/**
+ * Executes the provided command, piping the parent process' stderr, updating
+ * the error to process if stderr is not empty, and returns process object.
+ *
+ * @param {string} cmd
```
Comment (rsimha) [inline build-system/exec.js]: `@return {!Object}`

**Pull request merged**